include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -168,6 +168,12 @@ ERROR(error_formatting_multiple_file_ranges,none,
 ERROR(error_formatting_invalid_range,none,
   "file range is invalid", ())
 
+ERROR(invalid_conditional_compilation_flag,none,
+      "invalid conditional compilation flag in '-D%0'", (StringRef))
```

Comment:
Hm. In C the convention is to write `-DFOO`, but I'm not sure Swift should be encouraging that. `-D FOO` is equivalent and a bit easier to read.

The error message should also explain what the problem is, perhaps something like "conditional compilation flags must be valid Swift identifiers (rather than '%0')"?